src/braket/circuits/circuit.py

```diff
@@ -643,6 +657,69 @@ def add_verbatim_box(
             self._has_compiler_directives = True
         return self
 
+    def measure(self, target_qubits: Optional[np.ndarray] | Optional[int] = None) -> Circuit:
+        """
+        Add a `measure` operator to `self` ensuring only the target qubits are measured.
+
+        Args:
+            target_qubits (Optional[ndarray] | Optional[int]): target qubits to measure.
+                Default=None
+
+        Returns:
+            Circuit: self
+
+        Raises:
+            IndexError: If `circuit` has no qubits.
+            IndexError: If target qubits are not within the range of the current circuit.
+
+        Examples:
+            >>> circ = Circuit.h(0).cnot(0, 1).measure([0])
+            >>> circ.print(list(circ.instructions))
+            [Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(0)]),
+            Instruction('operator': CNot('qubit_count': 2), 'target': QubitSet([Qubit(0),
+                Qubit(1)]),
+            Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(2)]),
+            Instruction('operator': Measure, 'target': QubitSet([Qubit(0)])]
+        """
+        # check whether measuring an empty circuit
+        if not self.qubits:
+            raise IndexError("Cannot measure an empty circuit.")
+
+        if type(target_qubits) is int:
+            target_qubits = [target_qubits]
+
+        # Check that the target qubits are on the circuit
+        if target_qubits and not all(qubit in self.qubits for qubit in target_qubits):
+            raise IndexError("Target qubits must be within the range of the current circuit.")
+
+        # Check if result types are added on the circuit
+        if self.result_types:
+            raise ValueError("a circuit cannot contain both measure instructions and result types.")
+
+        if target_qubits:
+            self.add_instruction(
+                Instruction(
+                    operator=Measure(
+                        qubit_count=len(target_qubits), ascii_symbols=["M"] * len(target_qubits)
+                    ),
+                    target=target_qubits,
+                )
+            )
```

Comment:
If `Measure` becomes a single-qubit operation (per the other comment thread), then I think this would change to be adding a list of instructions, one for each qubit in `target_qubits`.

Reply:
Updated this to add a list of instruction since measure is now a single-qubit operation.

Reply:
Awesome!